src/main/java/net/tlabs/tablesaw/parquet/TablesawReadSupport.java

```diff
@@ -60,185 +60,153 @@
 
 public class TablesawReadSupport extends ReadSupport<Row> {
 
-  private final TablesawParquetReadOptions options;
-  private Table table = null;
-
-  public TablesawReadSupport(final TablesawParquetReadOptions options) {
-    super();
-    this.options = options;
-  }
-
-  @Override
-  public ReadContext init(final InitContext context) {
-    return new ReadContext(context.getFileSchema());
-  }
-
-  @Override
-  public RecordMaterializer<Row> prepareForRead(
-      final Configuration configuration,
-      final Map<String, String> keyValueMetaData,
-      final MessageType fileSchema,
-      final ReadContext readContext) {
-    this.table = createTable(fileSchema, this.options);
-    this.table.setName(this.options.tableName());
-    return new TablesawRecordMaterializer(this.table, fileSchema, this.options);
-  }
-
-  private Table createTable(final MessageType schema, final TablesawParquetReadOptions options) {
-    return Table.create(
-        schema.getFields().stream()
+    private final TablesawParquetReadOptions options;
+    private Table table = null;
+
+    public TablesawReadSupport(final TablesawParquetReadOptions options) {
+        super();
+        this.options = options;
+    }
+
+    @Override
+    public ReadContext init(final InitContext context) {
+        return new ReadContext(context.getFileSchema());
+    }
+
+    @Override
+    public RecordMaterializer<Row> prepareForRead(final Configuration configuration,
+            final Map<String, String> keyValueMetaData, final MessageType fileSchema, final ReadContext readContext) {
+        this.table = createTable(fileSchema, this.options);
+        this.table.setName(this.options.tableName());
+        return new TablesawRecordMaterializer(this.table, fileSchema, this.options);
+    }
+
+    private Table createTable(final MessageType schema, final TablesawParquetReadOptions options) {
+        return Table.create(
+            schema.getFields().stream()
             .map(f -> createColumn(f, options))
             .filter(Objects::nonNull)
             .collect(Collectors.toList()));
-  }
-
-  private Column<?> createColumn(final Type field, final TablesawParquetReadOptions options) {
-    final String name = field.getName();
-    if (field.isPrimitive() && !field.isRepetition(Repetition.REPEATED)) {
-      switch (field.asPrimitiveType().getPrimitiveTypeName()) {
-        case BOOLEAN:
-          return BooleanColumn.create(name);
-        case INT32:
-          return Optional.ofNullable(field.getLogicalTypeAnnotation())
-              .flatMap(
-                  a ->
-                      a.accept(
-                          new LogicalTypeAnnotationVisitor<Column<?>>() {
+    }
+
+    private Column<?> createColumn(final Type field, final TablesawParquetReadOptions options) {
```

Comment:
static

Reply:
side note: this method is too long and should be revisited

Reply:
Yes, there are a few methods that need refactoring...

Reply:
Made static too.

Reply:
FYI the Sonar code smells are about these methods ...